Android/ZensoftHRCRM/app/src/main/java/com/erkprog/zensofthrcrm/ui/vacancies/vacanciesList/VacanciesAdapter.java

```diff
@@ -0,0 +1,95 @@
+package com.erkprog.zensofthrcrm.ui.vacancies.vacanciesList;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import com.erkprog.zensofthrcrm.R;
+import com.erkprog.zensofthrcrm.data.entity.Vacancy;
+
+import java.util.List;
+
+public class VacanciesAdapter extends RecyclerView.Adapter<VacanciesAdapter.VacancyViewHolder> {
+
+  private List<Vacancy> mVacancies;
+  private Context mContext;
+  private OnItemClickListener mListener;
+
+  public void setOnItemClickListener(OnItemClickListener listener) {
+    mListener = listener;
+  }
+
+  public VacanciesAdapter(Context context, List<Vacancy> vacancies) {
+    mVacancies = vacancies;
+    mContext = context;
+  }
+
+  public void loadNewData(List<Vacancy> data) {
+    mVacancies = data;
+    this.notifyDataSetChanged();
+  }
+
+  @NonNull
+  @Override
+  public VacancyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+    View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.vacancy_item, parent,
+        false);
+    return new VacancyViewHolder(view, mListener);
+  }
+
+  @Override
+  public void onBindViewHolder(@NonNull VacancyViewHolder holder, int position) {
+    Vacancy vacancy = mVacancies.get(position);
+
+    holder.name.setText(vacancy.getName() != null ? vacancy.getName() : "");
```

Comment:
Add check if vacancy is not null, thus we do not need to check for null each item